### ISSUE TITLE
Cemgil Scores > 1

### ISSUE THREAD

**Issue by SebastianPoell** [CPJKU/madmom#414]
### The issue

Hi there, using simple thresholding as a beat detection method can represent a viable baseline when evaluating different beat trackers. By its nature, thresholding tends to gather multiple detections around an annotation. However, this leads to Cemgil scores > 1 and thus, usually ranks thresholding higher than all other algorithms. The reason is that activations and detections are swapped (in contrast to the original implementation). This is also commented in the code:

https://github.com/CPJKU/madmom/blob/41155f4dd876e8ba45940398a91046c07a271140/madmom/evaluation/beats.py#L473-L476

I guess, the swapping was done to prevent confusion about the parameter namings? For a 'normal' usecase this should work fine but I'm currious to whether we should rethink that swapping...

### Steps needed to reproduce the behaviour

Just evaluate this example. This results in Cemgil = 1.233.

[detections.txt](https://github.com/CPJKU/madmom/files/2834583/detections.txt)
[annotations.txt](https://github.com/CPJKU/madmom/files/2834584/annotations.txt)

**Comment by superbock** [comment]
Yes, that's definitely not the expected behaviour — but I am not sure if it is a bug with only Cemgil's metric.

Swapping annotations and detections reduces Cemgil's score to 0.601, which is way more reasonable. However, I am not sure at all if the other metrics behave as intended. Since your example has way more detections than annotations, also information gain is also pretty close to it's maximum. And even simple metrics like F-measure should be low given that many false positive detections. So it looks more like we are generally unable to handle that many detections.

I think it is safest to do proper peak picking after thresholding. Have you tried `features.onsets.peak_picking`? It does basically thresholding with local maxima peak picking.

P.S. I am not sure why I was swapping them in the first place, so I have to rethink the whole issue. Also I have to compare the results with `mir_eval` to see how their implementation behaves.

**Comment by SebastianPoell** [comment]
Yeah, you are right - where I also noticed it was P-Score (although it never went > 1). Hmm, maybe we should rename this issue?

Yes, In most cases, peak picking fixes it completely. However, in the worst case (online networks/online algorithms) I'm still getting ~3% difference for P-Score and Cemgil depending on the swapping. After all, this an edge case, but probably still worth a closer look 🤷‍♂️
